common/Path.cc

```diff
@@ -0,0 +1,107 @@
+#include "common/Path.h"
+
+namespace sorbet {
+using namespace std;
+
+#if defined(_WIN32) || defined(_WIN64)
+constexpr char file_separator = '\\';
+#else
+constexpr char file_separator = '/';
+#endif
+
+/**
+ * A simple naive implementation of Path for platform that do not have <filesystem> yet
+ */
+
+template <> BasePath<string>::BasePath(const string_view path) : internal_path(std::string(path)) {}
+
+template <> BasePath<string> &BasePath<string>::replace_filename(const string_view filename) {
```

Comment:
We use camelCase, btw, ~everywhere. We used to have a mix of snake and camel.

Reply:
Here I model the API to be the same as <filesystem> itself but I'll change to camelCase